### PR TITLE
Replace gtLclNum with GetLclNum()

### PR DESCRIPTION
**Pull request by franksinankaya**
@am11  @jkotas @janvorli @sandreenko

### CODE REVIEW

**Comment by sandreenko** [comment]
Looks like there is a failure with this change:
```
Assert failure(PID 3588 [0x00000e04], Thread: 2996 [0x0bb4]): Assertion failed '!m_lsra->isRegCandidate(varDsc)' in 'System.Numerics.Vector:Narrow(struct,struct):struct' (IL size 76)

    File: f:\workspace.2\_work\1\s\src\jit\lower.h Line: 366
    Image: F:\workspace.2\_work\1\s\bin\Product\Windows_NT.arm64.Checked\x64\crossgen.exe
```

I do not see a change in this PR that could cause it, but I have not seen this failure before, so it could be a regression. I will rerun the testing.

cc @dotnet/jit-contrib

File: src/jit/importer.cpp [hunk 1]
```diff
@@ -9064,7 +9065,7 @@ GenTree* Compiler::impFixupStructReturnType(GenTree* op, CORINFO_CLASS_HANDLE re
     {
         // It is possible that we now have a lclVar of scalar type.
         // If so, don't transform it to GT_LCL_FLD.
-        if (lvaTable[op->AsLclVar()->gtLclNum].lvType != info.compRetNativeType)
+        if (varTypeIsStruct(lvaTable[op->AsLclVar()->GetLclNum()].lvType))
```
Comment (franksinankaya) [inline src/jit/importer.cpp]: this looks like a rebase issue.

**Comment by franksinankaya** [comment]
@sandreenko : yet another permission issue. Can you kickstart the test again?

**Comment by franksinankaya** [comment]
@dotnet-bot test coreclr-ci

**Comment by sandreenko** [comment]
the test run looks good, the failures are from https://github.com/dotnet/coreclr/pull/26551, I am going to merge then without retesting (or you will have to rebase your changes to get the fix for the regression).

**Pull request merged**